pxr/imaging/hgiMetal/texture.mm

```diff
@@ -95,7 +103,11 @@
         // to our original, private texture.
 
         // Modify texture descriptor to describe the temp texture.
+#if defined(ARCH_OS_OSX)
```

Comment:
could we have a comment here about why the Mode is not the same on desktop and embedded? I realize Managed isn't available on iOS but why aren't both Shared?

Reply:
I'll double check with Thor if there's any other reason, but afaik it's because Managed is required for older Macs with a dGPU (AMD/NVIDIA but not Intel).

A better check would probably be to use [hasUnifiedMemory](https://developer.apple.com/documentation/metal/mtldevice/3229025-hasunifiedmemory), instead of the ifdef. 

However, I am torn on whether its best to leave Mac behaviour alone for now and address it in a future PR, so that we can keep the scope of this PR tight?

Reply:
Checked with Thor, and it is the reason above. Anyway, I'll wait to hear back from you before I add a comment or just swap to the unifiedMemory check which should then be self explanatory.

Reply:
tbh I'd prefer we switch to the check because I worry that such a minor detail might just slip away for more months. @davidgyu any thought on this?

Reply:
Fair enough. I made the change after discussing with a few folks who seem confident it shouldn't cause issues.
Unfortunately I do not have a Mac with an intel gpu to test on, but it seems to be working fine on Apple Silicon Macs, so I think we should be good.

Reply:
I like that suggestion @meshula. I might be able to check for correct behavior on my stash of old macOS systems with Intel and AMD GPUs.